rclcpp/test/rclcpp/test_create_timer.cpp

```diff
@@ -61,3 +61,40 @@ TEST(TestCreateTimer, call_with_node_wrapper_compiles)
     []() {});
   rclcpp::shutdown();
 }
+
+TEST(TestCreateTimer, call_wall_timer_with_bad_arguments)
```

Comment:
@brawner consider adding a test that forces a throw on `Timer` construction using a bad clock.

Reply:
Unfortunately, I think that's harder than it sounds. The constructor of WallTimer uses RCL_STEADY_CLOCK, but GenericTimer can use an arbitrary clock type. However, the constructor of Clock calls `rcl_clock_init`, which will validate it's one argument, `clock_type`. So I'm not sure it's feasible to pass in a bad clock without mocking the rclcpp::Clock class.

It's possible I'm missing something, do you have any suggestions along these lines?

Reply:
I think [`RCL_CLOCK_UNINITIALIZED`](https://github.com/ros2/rcl/blob/f84592fb2f6c93cb25dd4ca59b16858cf0d9abdf/rcl/include/rcl/time.h#L61) may serve your purposes.

Reply:
Added this test to test_timer.cpp